src/app/clusters/door-lock-server/door-lock-server.h

```diff
@@ -180,6 +180,17 @@ class DoorLockServer
         return GetFeatures(endpointId).Has(DoorLockFeature::kUser) && SupportsAnyCredential(endpointId);
     }
 
+    inline void ResetWrongCodeEntryAttempts(chip::EndpointId endpointId)
```

Comment:
Is this public API that consumers are expected to call?  I would have expected this to be called from SendLockOperationEvent as needed, so that consumers don't have to do that manually....

Reply:
Yes thats the intention. For the pinpad use case for e.g. the app doesn't send us the wrong entry info. so the app has to track the wrong code and call the HandleWrongCodeEntry or ResetWrongCodeEntryAttempts if  a correct pincode is entered. This was my understanding when talking to Joshua. I will cover this in the issue to track usage. Please let me know if we need to discuss this further

Reply:
My point is: the app needs to be calling SendLockOperationEvent anyway, unless it manually sends those events.  So what I would propose is:

1. SendLockOperationEvent handles this automatically.
2. We have a public API for this too.
3. We clearly document that you either send events yourself and call the reset API, or that you just call SendLockOperationEvent and it handles things.

Reply:
gotcha. I added a reset in SetLockState https://github.com/project-chip/connectedhomeip/pull/25536/commits/f41610c4fcce7ebeac8d3ef32404ca9a5d34ae35#diff-ba4731e35f6cf29195b5ca8a6c50f6ed2b1730e9c492ae6453d02ac34e9f0f74R129. Are you saying that apps will call SendLockOperationEvent and not call setLockState. If thats true, I will add the reset code there as well. Please lmk. Will add documentation.

Reply:
> my point is: the app needs to be calling SendLockOperationEvent anyway

I meant SetLockState, not SendLockOperationEvent....

Reply:
done